lib/segment/src/vector_storage/query_scorer/discovery_query_scorer.rs

```diff
@@ -0,0 +1,53 @@
+use std::marker::PhantomData;
+
+use common::types::{PointOffsetType, ScoreType};
+
+use crate::data_types::vectors::{VectorElementType, VectorType};
+use crate::spaces::metric::Metric;
+use crate::vector_storage::query::discovery_query::DiscoveryQuery;
+use crate::vector_storage::query_scorer::QueryScorer;
+use crate::vector_storage::VectorStorage;
+
+// TODO(luis): maybe find a way to generalize DiscoveryQueryScorer and RecoQueryScorer by
```

Comment:
**Note:** `discovery_query_scorer.rs` and `quantized_discovery_query_scorer.rs` are basically copy-pasted from `reco_query_scorer.rs` and `quantized_reco_query_scorer.rs` respectively. It may be a good idea to refactor them into a `custom_query_scorer` or similar. 

The crux is that I couldn't figure out how to make the `transform` function to fit nicely as a trait function. But if we solve that, we can get rid of this code repetition.